x-pack/qa/usage/src/javaRestTest/java/org/elasticsearch/xpack/usage/XPackUsageIT.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.usage;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.test.rest.ESRestTestCase;
+
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+/*
+ * This class is meant to test the _xpack/usage API. That API pulls in code from various plugins so it cannot be integration tested
+ * within x-pack/plugin/core.
+ */
+public class XPackUsageIT extends ESRestTestCase {
```

Comment:
Since this is testing only DLM should we make the name a bit more specific?

Reply:
I named the test method `testDataLifecycle`. But I named the test `XPackUsageIT` so that people could add other xpack usage tests here over time. I had hoped to find an XPackUsageIT that I could just add my method to, so I figured someone else might in the future.

On that note though, this is a very expensive test -- it can take about 30 seconds -- and I'm not sure it's actually worth the cost. We have similar (not quite as realistic) coverage in DataLifecycleUsageTransportActionIT (~10 seconds), and serialization coverage in DataLifecycleFeatureSetUsageTests (fast). I'm not sure this test is worth the cost for such simple code with coverage elsewhere.

Reply:
I see the concern. Can we have a yaml test like we had for the health API?